scripts/all-core.sh

```diff
@@ -937,6 +941,15 @@ run_component () {
         "${dd_cmd[@]}"
     fi
 
+    # Since building TF-PSA-Crypto is out of source, we cannot identify if we
+    # are in TF-PSA-Crypto repository. We cache the result, which is 0 for
+    # success.
+    in_tf_psa_crypto_repo
+    running_tf_psa_crypto_test="$?"
+    if [ "$running_tf_psa_crypto_test" -eq 0 ]; then
+        pre_create_tf_psa_crypto_out_of_source_directory
+    fi
```

Comment:
We run the script with `set -e` active. This means that if any command exit with a nonzero status, the script exits. Note that [this has a lot of limitations and pitfalls](https://mywiki.wooledge.org/BashFAQ/105), but those limitations aren't a concern here. Also, we run parts of the script with an EXIT trap which catches some here, but that too is not relevant at this particular point in the script.

When `all.sh` is running in the mbedtls repository, the command `in_tf_psa_crypto_repo` exits with the status 1. This causes the whole script to exit with 1.

Here the fix is to put the command directly in an if statement.

```suggestion
    if in_tf_psa_crypto_repo; then
        running_tf_psa_crypto_test=1
        pre_create_tf_psa_crypto_out_of_source_directory
    fi
```

Reply:
Thank you for saving me that debugging headache, that makes a lot more sense to me now.